File: EXAMEN/01.SERVIDOR/WS_JAVA_REST_Comercializadora/src/main/java/ec/edu/monster/model/Factura.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.monster.model;

/**
 *
 * @author danie
 */

import jakarta.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;

@Entity
@Table(name = "factura")
public class Factura {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private LocalDate fecha;

    @Column(name = "cedula_cliente", nullable = false, length = 10)
    private String cedulaCliente;

    @Column(name = "nombre_cliente", nullable = false, length = 100)
    private String nombreCliente;

    @Column(name = "forma_pago", nullable = false, length = 15)
    private String formaPago; // EFECTIVO, CREDITO

    @Column(name = "total_bruto", nullable = false, precision = 10, scale = 2)
    private BigDecimal totalBruto;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal descuento;

    @Column(name = "total_neto", nullable = false, precision = 10, scale = 2)
    private BigDecimal totalNeto;

    @Column(name = "id_credito")
    private Long idCreditoBanquito;  // null si fue EFECTIVO

    @OneToMany(mappedBy = "factura", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<DetalleFactura> detalles;

    // getters & setters (al menos los básicos)
    // ...

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public LocalDate getFecha() {
        return fecha;
    }

    public void setFecha(LocalDate fecha) {
        this.fecha = fecha;
    }

    public String getCedulaCliente() {
        return cedulaCliente;
    }

    public void setCedulaCliente(String cedulaCliente) {
        this.cedulaCliente = cedulaCliente;
    }

    public String getNombreCliente() {
        return nombreCliente;
    }

    public void setNombreCliente(String nombreCliente) {
        this.nombreCliente = nombreCliente;
    }

    public String getFormaPago() {
        return formaPago;
    }

    public void setFormaPago(String formaPago) {
        this.formaPago = formaPago;
    }

    public BigDecimal getTotalBruto() {
        return totalBruto;
    }

    public void setTotalBruto(BigDecimal totalBruto) {
        this.totalBruto = totalBruto;
    }

    public BigDecimal getDescuento() {
        return descuento;
    }

    public void setDescuento(BigDecimal descuento) {
        this.descuento = descuento;
    }

    public BigDecimal getTotalNeto() {
        return totalNeto;
    }

    public void setTotalNeto(BigDecimal totalNeto) {
        this.totalNeto = totalNeto;
    }

    public Long getIdCreditoBanquito() {
        return idCreditoBanquito;
    }

    public void setIdCreditoBanquito(Long idCreditoBanquito) {
        this.idCreditoBanquito = idCreditoBanquito;
    }

    public List<DetalleFactura> getDetalles() {
        return detalles;
    }

    public void setDetalles(List<DetalleFactura> detalles) {
        this.detalles = detalles;
    }
}

File: PROYECTO/01.SERVIDOR/WS_DOTNET_SOAP_Comercializadora/ec.edu.model/Factura.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Numerics;

namespace WS_DOTNET_SOAP_Comercializadora.ec.edu.monster.model
{
    [Table("factura")]
    public class Factura
    {
        [Key]
        [Column("id")]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; }

        [Column("fecha")]
        public DateTime Fecha { get; set; }

        [Required]
        [Column("cedula_cliente")]
        [StringLength(10)]
        public string CedulaCliente { get; set; }

        [Required]
        [Column("nombre_cliente")]
        [StringLength(100)]
        public string NombreCliente { get; set; }

        [Required]
        [Column("forma_pago")]
        [StringLength(15)]
        public string FormaPago { get; set; } // EFECTIVO / CREDITO

        [Required]
        [Column("total_bruto", TypeName = "decimal")]
        public decimal TotalBruto { get; set; }

        [Required]
        [Column("descuento", TypeName = "decimal")]
        public decimal Descuento { get; set; }

        [Required]
        [Column("total_neto", TypeName = "decimal")]
        public decimal TotalNeto { get; set; }

        [Column("id_credito")]
        public long? IdCreditoBanquito { get; set; }

        // RELACIÓN 1-N
        public ICollection<DetalleFactura> Detalles { get; set; }
    }
}
